WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -175,14 +191,26 @@ final class ProductSelectorViewModel: ObservableObject {
     /// Select a product to add to the order
     ///
     func selectProduct(_ productID: Int64) {
+        // If it's synching and there's an attempt to select a product, return early
+        if shouldShowScrollIndicator {
+            return
+        }
```

Comment:
I don't think this property `shouldShowScrollIndicator` has anything to do with product selection. Commenting out this code works the same way. I'd suggest removing it.

Reply:
Thanks for digging into this. I see we're calling it through sync, but not when we do product selection but whether the product list is loading more content. I removed it here https://github.com/woocommerce/woocommerce-ios/pull/8935/commits/25b7bdfab3c4142ffa2d0d9f17a33f681049cdeb , and confirmed that we still block the UI when the product is being selected, disallowing from selecting another one ( which is what we want ). 

Something that can be confusing when testing this and checking the Order after selecting a bunch of products is what I mentioned [here](https://github.com/woocommerce/woocommerce-ios/pull/8935#issuecomment-1449881698) regarding both selecting and unselecting a Product adds such Product to an Order (unselecting does not remove them from the Order), this will be dealt with on a https://github.com/woocommerce/woocommerce-ios/pull/8943 and I can test further if there's any issue blocking the UI, or revisit how we reflect this to the end-user.